crates/nu-command/src/filesystem/cd.rs

```diff
@@ -246,14 +247,43 @@ fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
             use std::os::unix::fs::MetadataExt;
             let bits = metadata.mode();
             let has_bit = |bit| bits & bit == bit;
-            let current_user = users::get_current_uid();
-            if current_user == 0 {
+            let current_user_uid = users::get_current_uid();
+            if current_user_uid == 0 {
                 return PermissionResult::PermissionOk;
             }
-            let current_group = users::get_current_gid();
+
+            let current_user_gid = users::get_current_gid();
+            let mut current_user_groups = match users::get_current_username() {
+                None => {
+                    return PermissionResult::PermissionDenied("Unable to get current user name")
```

Comment:
Generally, I don't think we need to return these `PermissionResult::PermissionDenied` value, if we can't get username(it's really rare, I'm failed to think when it'd fail), we can just make `current_user_groups` an empty Vec, then if user group_owner don't in user_groups, just return an original PermissionDenied message

If we don't care about these error, we can lead to simpler implementation:
```rust
let current_user_groups = users::get_current_username()
    .map(|name| {
        users::get_user_groups(&name, current_user_gid)
            .map(|mut groups| {
                // comments about https://github.com/ogham/rust-users/issues/44 here
                if groups.len() >= 2 {
                    groups.pop();
                }
                groups
            })
            .unwrap_or_default()
    })
    .unwrap_or_default();
```